### PR TITLE
Parallel click test

### PR DESCRIPTION
**Pull request by codingtwinky**
Add parallel click test for speeding up dev efforts.

**Tests may time out depends on the parallelization level and resource allocation.** tried to minimize those variables via removing some of the unnecessary heavy network IOs but it is not gurantted.

### CODE REVIEW

File: components/commit/commit.js [hunk 1]
```diff
@@ -32,9 +32,7 @@ function CommitViewModel(gitNode) {
   this.numberOfAddedLines = ko.observable();
   this.numberOfRemovedLines = ko.observable();
   this.authorGravatar = ko.computed(function() {
-    var lowerCaseEmail = self.authorEmail();
-    lowerCaseEmail = lowerCaseEmail ? lowerCaseEmail.trim().toLowerCase() : lowerCaseEmail;
-    return md5(lowerCaseEmail);
+    return md5((self.authorEmail() || "").trim().toLowerCase);
```
Comment (campersau) [inline components/commit/commit.js]: `.toLowerCase()`

File: Gruntfile.js [hunk 2]
```diff
@@ -382,6 +382,59 @@ module.exports = function(grunt) {
...
+  grunt.registerTask('clickParallel', 'Parallelized click tests.', function() {
+    var done = this.async();
+
+    fs.readdirAsync('./nmclicktests')
+      .then(function(files) {
+        return files.filter(function(file) {
+          return file.startsWith("spec.");
+        });
+      }).then(function(tests) {
+        grunt.log.writeln('Running click tests in parallel... (this will take a while...) \t');
```
Comment (campersau) [inline Gruntfile.js]: I think `\t` is not necessary here?

File: components/commit/commit.js (hunk 1 above)
Comment (codingtwinky) [inline components/commit/commit.js]: somebody has been working on scala too much... >.<

**Comment by codingtwinky** [comment]
Thanks for the review!!  you are a life saver!  

Also, I don't know if you do read some of my other PRs, could you please comment even if you don't catch anything?  Although I'm sure all of my PRs are littered with mistakes.  But if you could confirm however informally, it would make me feel so much better.

**Comment by campersau** [comment]
@codingtwinky I have reviewed all of your PRs. In the future I will comment even if I can't find something. <sub>Which does not necessarily mean that there are no bugs :smirk:</sub>

**Comment by codingtwinky** [comment]
Oh for sure, I'd just be happy knowing that there is some I can share ~~blame~ joy of programming with.

File: nmclicktests/environment.js [hunk 3]
```diff
@@ -218,7 +218,7 @@ class Environment {
     return this.getPort()
       .then(() => this.startServer())
       .then(() => this.ensureStarted())
-      .catch((err) => { throw new Error("Cannot confirm ungit start!!"); })
+      .catch((err) => { console.log(err); throw new Error("Cannot confirm ungit start!!", err); })
```
Comment (campersau) [inline nmclicktests/environment.js]: In JavaScript there is no `Error` constructor which takes the "inner" / "previous" error.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error
https://nodejs.org/api/errors.html#errors_new_error_message

**Pull request merged**